cupy/cuda/compiler.py

```diff
@@ -89,21 +89,23 @@ def _run_cc(cmd, cwd, backend, log_stream=None):
 
 @_util.memoize()
 def _get_extra_path_for_msvc():
-    import distutils.spawn
-    cl_exe = distutils.spawn.find_executable('cl.exe')
+    import shutil
+    cl_exe = shutil.which('cl.exe')
     if cl_exe:
         # The compiler is already on PATH, no extra path needed.
         return None
 
-    from distutils import msvc9compiler
-    vcvarsall_bat = msvc9compiler.find_vcvarsall(
-        msvc9compiler.get_build_version())
-    if not vcvarsall_bat:
+    from setuptools import msvc
+    from setuptools.errors import DistutilsError
```

Comment:
It seems `setuptools.errors` is only available in setuptools v42.0+. In my environment, the default one that comes with Python 3.8 for Windows was 41.2.0, so this line failed. Until we decide a policy on which setuptools versions to support (we should do this once we fully migrate from distutils), I'm negative to use this module. So, could you use `distutils.errors.DistutilsPlatformError` instead, as in your original code? :pray: